internal/core/docker_host.go

```diff
@@ -67,9 +84,16 @@ func DefaultGatewayIP() (string, error) {
 //  5. Docker host from the "docker.host" property in the ~/.testcontainers.properties file.
 //  6. Rootless docker socket path.
 //  7. Else, the default Docker socket including schema will be returned.
-func ExtractDockerHost(ctx context.Context) string {
+func ExtractDockerHost(ctx context.Context, callbackChecks ...func(host string) error) string {
 	dockerHostOnce.Do(func() {
-		dockerHostCache = extractDockerHost(ctx)
+		// the tests can pass a custom callback check to avoid the Docker client creation
+		// and to avoid changing the code base, passing zero callbacks will mean the default one,
+		// which checks that the host is reachable.
+		if len(callbackChecks) == 0 {
+			callbackChecks = append(callbackChecks, defaultCallbackCheckFn)
```

Comment:
suggestion: if the above is correct then we need to:
```suggestion
			callbackChecks = append(callbackChecks, defaultCallbackCheck)
```